packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -123,21 +123,30 @@ export class Relayer {
 
     // Build dictionary of pending relays keyed by l1 token and deposit hash. getPendingRelays() filters
     // out Finalized relays.
-    const pendingRelays: Relay[] = this.l1Client.getPendingRelayedDeposits();
-    if (pendingRelays.length == 0) {
+    const pendingRelays: { [key: string]: Relay[] } = this.l1Client.getPendingRelayedDepositsGroupedByL1Token();
+    if (Object.keys(pendingRelays).length == 0) {
       this.logger.debug({ at: "AcrossRelayer#Disputer", message: "No pending relays" });
       return;
     }
-    this.logger.debug({ at: "AcrossRelayer#Disputer", message: `Processing ${pendingRelays.length} pending relays` });
-
-    for (const relay of pendingRelays) {
+    this.logger.debug({
+      at: "AcrossRelayer#Disputer",
+      message: `Processing ${Object.keys(pendingRelays).length} l1 token pending relays`,
+    });
+    for (const l1Token of Object.keys(pendingRelays)) {
```

Comment:
this change (and the introduction of `getPendingRelayedDepositsGroupedByL1Token`) is done to ensure that we only send the batch disputes to one contract _at a time_. i.e consider if both the USDC and WETH pool have disputes. we cant send these in a batch as the destination contract is different between the two. The implementation here now sends them based off a grouping of `BridgePool`.

Reply:
You should add this as a comment to the code to make sure no one accidentally introduces the possiiblity that batches can be sent to multiple contracts